// capacitor-moengage-core/dist/plugin.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@capacitor/core');

/**
 * User attribute Gender options
 */
exports.MoEUserGender = void 0;
(function (MoEUserGender) {
    MoEUserGender["MALE"] = "male";
    MoEUserGender["FEMALE"] = "female";
    MoEUserGender["OTHER"] = "other";
})(exports.MoEUserGender || (exports.MoEUserGender = {}));
/**
 * Status of the application.
 */
exports.MoEAppStatus = void 0;
(function (MoEAppStatus) {
    /**
     * Fresh installation of the app instance.
     */
    MoEAppStatus["INSTALL"] = "install";
    /**
     * App was already present and user has updated the app.
     */
    MoEAppStatus["UPDATE"] = "update";
})(exports.MoEAppStatus || (exports.MoEAppStatus = {}));
/**
 * InApp Action type
 */
exports.MoEInAppActionType = void 0;
(function (MoEInAppActionType) {
    MoEInAppActionType["NAVIGATION"] = "navigation";
    MoEInAppActionType["CUSTOM"] = "custom";
})(exports.MoEInAppActionType || (exports.MoEInAppActionType = {}));
/**
 * Platform Data
 */
exports.MoEPlatform = void 0;
(function (MoEPlatform) {
    MoEPlatform["iOS"] = "iOS";
    MoEPlatform["ANDROID"] = "android";
})(exports.MoEPlatform || (exports.MoEPlatform = {}));
/**
 * Supported push services
 */
exports.MoEPushService = void 0;
(function (MoEPushService) {
    MoEPushService[MoEPushService["APNS"] = 0] = "APNS";
    MoEPushService[MoEPushService["FCM"] = 1] = "FCM";
    MoEPushService[MoEPushService["MI_PUSH"] = 2] = "MI_PUSH";
    MoEPushService[MoEPushService["PUSH_KIT"] = 3] = "PUSH_KIT";
})(exports.MoEPushService || (exports.MoEPushService = {}));
/**
 * Permission Type
 */
exports.MoEPermissionType = void 0;
(function (MoEPermissionType) {
    MoEPermissionType["PUSH"] = "push";
})(exports.MoEPermissionType || (exports.MoEPermissionType = {}));

const MoECapacitorCore = core.registerPlugin('MoECapacitorCore', {});

exports.MoECapacitorCore = MoECapacitorCore;
